tests/performance_meter/test_performance.py

```diff
@@ -0,0 +1,221 @@
+import logging
+import asyncio
+import time
+import random
+import statistics
+import pytest
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+
+
+async def async_command(duthost, command):
+    return duthost.command(command)
+
+
+async def async_command_ignore_errors(duthost, command):
+    try:
+        return duthost.command(command, module_ignore_errors=True)
+    except Exception:
+        return
+
+
+# Defining an op.
+# An op is seperated into 2 parts by yield. The first part is setup
+# and happens before checking for success criteria, and the second
+# part is cleanup and happens after checking for success criteria
+# and the goal is to make sure that no change is leftover.
+# An op can be blocking or nonblocking, or combined, depending on
+# the need. For example, sometimes we want reboot to block until it
+# is successfully done before proceeding, or we want reboot to not
+# block at all to calculate how much time it takes to reboot.
+# If an op does not have yield in it, it will be treated as blocking.
+# Timing will only start after the first blocking part of operation
+# is over. The op should make sure op is started correctly and ended
+# correctly. If either part is unsuccessful, op should yeild False and
+# log the error, otherwise yielding True is expected.
```

Comment:
In short, the op defines the procedure:

1. OP - Do something
2. OP - yield or return
3. Start timer and success criteria checker
4. OP - Do something that we want to measure

Is this statement right?

Reply:
Yes